src/serialize.js

```diff
@@ -25,9 +25,16 @@ const removeDuplicateIncludes = included => {
   }, [])
 }
 
+const pickAttributes = (resource, attributeNames) => {
+  const attributes = attributeNames == null ? resource : pick(resource, attributeNames)
+  delete attributes.id
+
+  return attributes
+}
+
 const extractResourceInformation = (resource, attributeNames, relationshipNames) => {
   const identifier = extractIdentifier(resource)
-  const permittedAttributes = pick(resource, attributeNames)
+  const permittedAttributes = pickAttributes(resource, attributeNames)
```

Comment:
```suggestion
  const permittedAttributes = attributeNames == null
    ? omit(resource, ['id'])
    : pick(resource, attributeNames)
```

(or do that in the `pickAttributes` function above)